src/troubleshooting/content.md

```diff
@@ -0,0 +1,641 @@
+**TL;DR:** here's a diagram to help you debug your deployments in Kubernetes.
+
+![Flow chart to debug deployments in Kubernetes](diagram.png)
+
+When you wish to deploy an application in Kubernetes, you usually define three components:
+
+- a Deployment — which is a recipe for creating copies of your application called Pods
+- a Service — an internal load balancer that routes the traffic to Pods
+- an Ingress — a description of how the traffic should flow from outside the cluster to your Service.
+
+Here's a quick visual recap.
+
+```slideshow
+{
+  "description": "Kubernetes core concepts: Ingress, Service, Pod and Deployment.",
+  "slides": [
+    {
+      "image": "core-concepts-1.svg",
+      "description": "In Kubernetes your applications are exposed through two layers of load balancers: internal and external."
+    },
+    {
+      "image": "core-concepts-2.svg",
+      "description": "The internal load balancer is called Service, whereas the external one is called Ingress."
+    },
+    {
+      "image": "core-concepts-3.svg",
+      "description": "Pods are not deployed directly. Instead, the Deployment creates the Pods and whatches over them."
+    }
+  ]
+}
+```
+
+Assuming you wish to deploy a simple _Hello World_ application, the YAML for such application should look similar to this:
+
+```yaml|highlight=2,23,34|title=hello-world.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
```

Comment:
teaching idea, but please ignore me if stupid: when I explain a system in hungarian to friends, I always use a trick.  For every name that is optional and up to me, I choose a hungarian name, so that we can differentiate the system (english) from the customizable parts (hungarian).

In english we can achieve the same with "my"ing everything.  Or we could start using some company prefix for this (LK?)

```
  metadata:
    name: app
    labels:
      mytrack: canary (or lktrack: canary)
```

People always ask on the course: do I need a "name"?  Do I need an "app" under labels?

Reply:
I think this is an excellent idea that we could extend to the Academy @weibeld 
Any idea on how we should structure those prefixes? just `my-<something>`?

Reply:
I usually use `my-` or `my`. I think it makes sense to standardise this. It could even be `my-track: my-canary` to make it absolutely clear to beginners that these names are chosen by us and have no meaning to Kubernetes.